beacon_node/network/src/sync/range_sync/batch.rs

```diff
@@ -106,6 +106,8 @@ pub enum BatchState<T: EthSpec> {
     AwaitingValidation(Attempt),
     /// Intermediate state for inner state handling.
     Poisoned,
+    /// The batch is waiting for the execution layer to resume validation.
+    WaitingOnExecution,
```

Comment:
I still don't know why clippy does not pick this up, but `BatchState::AwaitingExecution` is not created anywhere. When we get a `stall_execution = true` we are using `BatchState::AwaitingDownload` without counting an additional failed attempt. I think this is the way to go and the new state is not necessary.

Reply:
Sorry, forgot to remove this one. I think clippy didn't pick this up because the enum variant was checked in match statements.